src/server/snapshot.cc

```diff
@@ -333,6 +320,17 @@ size_t SliceSnapshot::FlushSerialized(SerializerBase::FlushState flush_state) {
   uint64_t id = rec_id_++;
   DVLOG(2) << "Pushing " << id;
 
+  // FlushToSink can be quite slow for large values or due compression.
+  // In this case we sleep to avoid starvation of other fibers.
+  // Please note that we allocate the record id before we preempt in order to preserve ordering
+  // guarantees.
+  const uint64_t kSleepThreshold = base::CycleClock::Frequency() >> 15;  // ~30 usec
+  uint64_t elapsed = base::CycleClock::Now() - start;
+  if (elapsed > kSleepThreshold) {
+    // Balance the load between fibers.
+    ThisFiber::SleepFor(chrono::microseconds(200));
```

Comment:
why should we add the sleep here?
the next step is to move the data to the consumer , for replication call sink->write which is blocking and than another fiber will run

Reply:
it's about how much CPU time is taken by this fiber. if FlushSerialized took 200usec, and then this fiber suspends for 20usec to  send the data into a socket, then it de-facto occupies most of the CPU time. And this is what I see - I run debug populate RAND and when during full sync all the CPU is green at 100% and `perf top` shows it's busy compressing with LZ4. If I disable compression the throughput goes up.

This change factually helps to balance things - I am not saying it's perfect but I do not see how else we can balance cpu intensive tasks that use hundreds of microseconds of CPU time and maybe even millis.